src/conditions/mod.rs

```diff
@@ -9,6 +9,24 @@ pub mod is_metric;
 
 pub use check_fields::CheckFieldsConfig;
 
+/// A condition enum that can be optionally parsed without a `type` field, and
+/// defaults to a `check_fields` condition.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum DefaultedCondition {
+    FromType(Box<dyn ConditionConfig>),
+    NoTypeCondition(CheckFieldsConfig),
+}
+
+impl DefaultedCondition {
+    pub fn build(&self) -> crate::Result<Box<dyn Condition>> {
+        Ok(match self {
+            DefaultedCondition::FromType(c) => c.build()?,
+            DefaultedCondition::NoTypeCondition(c) => c.build()?,
+        })
+    }
+}
+
```

Comment:
And this can be removed, looks like it has been implemented since as https://github.com/timberio/vector/blob/master/src/conditions/mod.rs#L37